### PR TITLE
AEA-914 Generic Seller Skill Tests

### PR DESCRIPTION
**Pull request by 5A11**
This PR adds tests for the generic seller skill using the framework's skill testing tool.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/test_packages/test_skills/test_generic_seller/test_models.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+    def test_get_unregister_service_description(self):
+        """Test the get_unregister_service_description method of the GenericStrategy class."""
+        description = self.strategy.get_unregister_service_description()
+
+        assert type(description) == Description
+        assert description.data_model is AGENT_REMOVE_SERVICE_MODEL
+
+        assert "key" in description.values
+        assert description.values["key"] == "seller_service"
```
Comment (DavidMinarsch) [inline tests/test_packages/test_skills/test_generic_seller/test_models.py:130]:
```suggestion
        assert description.value.get("key", "") == "seller_service"
```

File: tests/test_packages/test_skills/test_generic_seller/test_behaviours.py [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+        has_attributes, error_str = self.message_has_attributes(
+            actual_message=self.get_message_from_outbox(),
+            message_type=OefSearchMessage,
+            performative=OefSearchMessage.Performative.REGISTER_SERVICE,
+            to=self.skill.skill_context.search_service_address,
+            sender=self.skill.skill_context.agent_address,
+            service_description=mocked_description_2,
+        )
+        assert has_attributes, error_str
+        mock_logger.assert_any_call(logging.INFO, "registering service on SOEF.")
```
Comment (DavidMinarsch) [inline tests/test_packages/test_skills/test_generic_seller/test_behaviours.py:170]: In tests you can also use private helper methods to reduce code duplication

**Review by DavidMinarsch — COMMENTED** [review]
Please update `tox.ini`:

```
commands = pytest -rfE --doctest-modules aea packages/fetchai/connections packages/fetchai/protocols packages/fetchai/skills/generic_buyer tests/ --cov-report=html --cov-report=xml --cov-report=term --cov-report=term-missing --cov=aea --cov=packages/fetchai/connections --cov=packages/fetchai/protocols --cov=packages/fetchai/skills/generic_buyer --cov=packages/fetchai/skills/generic_seller --cov-config=.coveragerc {posargs}
```

File: tests/test_packages/test_skills/test_generic_seller/test_models.py (hunk 1 above)
Comment (5A11) [reply tests/test_packages/test_skills/test_generic_seller/test_models.py:130]: Done

File: tests/test_packages/test_skills/test_generic_seller/test_behaviours.py [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+        quantity = self.get_quantity_in_outbox()
+        assert (
+            quantity == 3
+        ), f"Invalid number of messages in outbox. Expected 3. Found {quantity}."
+
+        # message 1
+        has_attributes, error_str = self.message_has_attributes(
+            actual_message=self.get_message_from_outbox(),
+            message_type=LedgerApiMessage,
+            performative=LedgerApiMessage.Performative.GET_BALANCE,
```
Comment (5A11) [inline tests/test_packages/test_skills/test_generic_seller/test_behaviours.py:109]: The following would help:

Taking out `XMessage.Performative` from `XMessage` class into its own independent class in the same module and renaming them as `XPerformative` (e.g. `FipaPerformative`). These can subclass a base Performative enum that resides in the same module as the base Message. The enum can then also have a property, which gives the XMessage class it is linked to. This way one can do:

```
performative.message_class
```

to access the message the performative belongs. This has several benefits, an immediate one being in the above function, you don't need to specify the message class. From a performative, its appropriate message_class is always accessible.

P.S. as a side note, it seems having inner classes in python is not a very encouraged practice, as opposed to other languages:
https://stackoverflow.com/a/2024583/1460594
https://stackoverflow.com/questions/719705/what-is-the-purpose-of-pythons-inner-classes/722175#722175

File: tests/test_packages/test_skills/test_generic_seller/test_behaviours.py (hunk 2 above)
Comment (5A11) [reply tests/test_packages/test_skills/test_generic_seller/test_behaviours.py]: Done

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**